tests/integration/simple_integration_test.go

```diff
@@ -0,0 +1,164 @@
+package integration_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"io/ioutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var (
+	namespace = "k8ssandra"
+	releaseName string
+	releaseNameCluster string
+)
+
+var _ = Describe("Install the cluster", func() {
+	// TODO This assumes the kind was started with correctly setup cluster. Could we automate it here to provider proper e2e?
+
+	BeforeSuite(func() {
+		kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+		options := &helm.Options{KubectlOptions: kubectlOptions}
+
+		// Add traefik repo and update repos
+		helm.RunHelmCommandAndGetOutputE(GinkgoT(), options, "repo", "add", "traefik", "https://helm.traefik.io/traefik")
+		helm.RunHelmCommandAndGetOutputE(GinkgoT(), options, "repo", "update")
+
+		// Deploy traefik
+		// helm install traefik traefik/traefik -n traefik --create-namespace -f docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml
+		valuesPath, _ := filepath.Abs("../../docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml")
+		helm.RunHelmCommandAndGetOutputE(GinkgoT(), options, "install", "traefik", "traefik/traefik", "-n", "traefik", "--create-namespace", "-f", valuesPath)
+	})
+
+	BeforeEach(func() {
+		kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+		k8s.CreateNamespace(GinkgoT(), kubectlOptions, "k8ssandra")
+	})
+
+	AfterEach(func() {
+		kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+		options := &helm.Options{KubectlOptions: kubectlOptions}
+
+		// Verify everything was actually deleted.. I don't think Service is
```

Comment:
Is this actually verifying that things are deleted?

Reply:
It's not, since we know it's not deleted (as there's no deletion job). Some of these comments are meant for "TODO".

Reply:
@burmanm 

[This module](https://github.com/k8ssandra/k8ssandra/blob/f90a29de0d19eab0838663a223e54ab6769b240e/tests/integration/util/lib.go) may help where we now have `Ginkgo` & `Omega` into the `Terratest` foundations for k8ssandra operator & cluster integration tests.

One approach may be that we iterate over the resource(s) scoped by namespace and use of test driven resource names.  

For now, I've been targeting the pods and releases.  If we need to go to `crd` and `service` levels is necessary, those should be doable as we can use similar patterns for verification of their existence or non-existence.

An example of using the integration test module:

```
By("Expecting to have the operator and cluster in deployed status")
Ω(IsReleaseDeployed(options.Operator, "k8ssandra")).Should(BeTrue())
Ω(IsReleaseDeployed(options.Cluster, "cassdc")).Should(BeTrue())

By("Expecting to have running cass-operator pod")
Ω(IsPodRunning(options.Operator, "cass-operator")).Should(BeTrue())
```

Reply:
@jeffbanks We definitely will have a need for a common test library. I do not want to get too caught focusing on it right now though. I fear doing so may be a bit of a premature optimization considering we currently have limited test automation right across the various repos. I would rather focus more on what scenarios we want to test and getting coverage in place. Then we can iterate and refactor common code as we go.

>For now, I've been targeting the pods and releases.

Take a top down approach. Start with thinking about the various custom resources we are creating.

* How do we verify that a `CassandraDatacenter` object is ready?
* How do we verify that a `Reaper` object is ready?
* How do we verify that a `Grafana` object is ready?
* How do we verify that a `Prometheus` object is ready?

Then let's think about the various resources that are charts are installing or not installing depending on what is enabled. We should able to get unit tests in place fairly easily to cover this stuff. 

We should also have unit tests to make sure labels are being set correctly.